interactives/interactives.go

```diff
@@ -142,7 +144,17 @@ func (c *Client) ListInteractives(ctx context.Context, userAuthToken, serviceAut
 		if err := q.Validate(); err != nil {
 			return List{}, err
 		}
-		uri = fmt.Sprintf("%s?offset=%d&limit=%d", uri, q.Offset, q.Limit)
+		uri = fmt.Sprintf("%s?offset=%d", uri, q.Offset)
+		if q.Limit > 0 {
+			uri = fmt.Sprintf("%s&limit=%d", uri, q.Limit)
+		}
+		if q.Filter != nil {
+			marshalled, jsonerr := json.Marshal(q.Filter)
+			if jsonerr != nil {
+				return List{}, jsonerr
+			}
+			uri = fmt.Sprintf("%s&filter=%s", uri, url.QueryEscape(string(marshalled)))
```

Comment:
not sure `QueryEscape` is needed - its encoded further down 😄  https://github.com/ONSdigital/dp-api-clients-go/blob/72c9fb08bebb1015a6302e8d399909441b9516d1/interactives/interactives.go#L213